students/jiwoo/users/views.py

```diff
@@ -1,53 +1,68 @@
-import json 
+import json, bcrypt, jwt
 
-from django.core.validators  import validate_email
-from django.core.exceptions  import ValidationError
 from django.http             import JsonResponse
-from django.db               import IntegrityError
 from django.views            import View 
 
 from users.models            import User
-from users.validations       import password_check, email_check
+from users.validations       import email_check, password_check, create_bcrypt, check_bcrypt, create_jwt
+from my_settings             import SECRET_KEY
+
 
 class SignupsView(View)  : 
-    def post(self, request): 
-        
+    def post(self, request):         
         try: 
             data         = json.loads(request.body)
             email        = data['email']
             password     = data['password']
             phone_number = data['phone_number']
             name         = data['name']
 
-            if not email_check(email):
+            if not email_check(email): 
                 return JsonResponse({"message":"INVALID_EMAIL_OR_PASSWORD"},status=400)
 
             if not password_check(password): 
                 return JsonResponse({"message":"INVALID_EMAIL_OR_PASSWORD"},status=400)
 
             if User.objects.filter(email=email).exists(): 
-                return JsonResponse({"message":"EMAIL_ALREADY_EXISTS"},status=400)
-            
+                return JsonResponse({"message":"EMAIL_ALREADY_EXISTS"},status=400)     
+
             if User.objects.filter(phone_number=phone_number).exists(): 
                 return JsonResponse({"message":"PHONE_NUMBER_ALREADY_EXISTS"},status=400)
 
             if User.objects.filter(name=name).exists(): 
                 return JsonResponse({"message":"NAME_ALREADY_EXISTS"},status=400)
-            
+
+            hashed_password  = create_bcrypt(password)
+            decoded_password = hashed_password.decode()
+
             User.objects.create(
                 email        = email,
-                password     = password,
+                password     = decoded_password,
                 phone_number = phone_number,
                 name         = name
             )
-            return JsonResponse({"message":"SUCCESS"},status = 201)
-        
-
+            return JsonResponse({"message":"SUCCESS"},status = 201)      
         except KeyError: 
             return JsonResponse({"MESSAGE":"KEY_ERROR"},status=400)
-        except ValidationError: 
-            return JsonResponse({"message":"VALIDATION_ERROR"},status=400)
-                
 
 
+class SigninsView(View)  : 
+    def post(self, request): 
+        try: 
+            data     = json.loads(request.body)
+            email    = data['email']
+            password = data['password']
+
+            if not User.objects.filter(email=email).exists(): 
+                return JsonResponse({"message":"INVALID_USER"},status=401)
 
+            hashed_password = User.objects.get(email=email).password.encode('utf-8')
```

Comment:
User.objects.get(email=email) 이 반복됩니다. 
user 라는 변수에 담아 사용하시는건 어떤가요?
get 메소드를 사용할 때마다 데이터베이스를 조회하게 되는데, 최초에 조회를 통한 인스턴스를 변수에 담아 사용하면 데이터베이스 조회량을 줄일 수 있을 것 같습니다.

```suggestion
        user = User.objects.get(email=email)
        
        ...
        hashed_password = user.password.encode('utf-8')
```

Reply:
변수에 담게되면 확실히 데이터베이스 조회량을 줄일수 있겠네요!! 변수로 바꿔보았습니다~!